mewarpx/mewarpx/utils_store/profileparser.py

```diff
@@ -0,0 +1,158 @@
+
+        for line in self.lines:
+            if "------" in line:
+                break
+            # these lines are of the form "Function()  ncalls  min  avg  max  max_percent"
+            # with more spaces than are here
+
+            # replace the multiple spaces with a single space
+            line = re.sub(" +", " ", line)
+            values = line.split(" ")
+
+            # work backwards through the values in order to capture function names with spaces.
+            max_percent = float(values[-1].replace("%", ""))
+            max_time = float(values[-2])
+            avg = float(values[-3])
+            min_time = float(values[-4])
+            n_calls = int(values[-5])
+            name = "_".join(values[0:-5])
+
+            dict = {
+                "frame" : {"name" : name},
+                "metrics" : {
+                    "n_calls" : n_calls,
+                    f"{section}_min" : min_time,
+                    f"{section}_avg" : avg,
+                    f"{section}_max" : max_time,
+                    "max_percent" : max_percent
+                }
+            }
+
+            if section == "excl":
+                self.exclusive_profiles.append(dict)
+            elif section == "incl":
+                self.inclusive_profiles.append(dict)
+
+            count = count + 1
+
+        return count
+
+    def write_file(self):
+        if not self.profile_dicts:
+            raise RuntimeError("write_file called before profile data was parsed!")
+
+        print("Writing profile data json file...")
+        with open(os.path.join(self.write_dir, "profile_data.json"), "w") as data_file:
+            data_file.write(json.dumps(self.profile_dicts, indent=4))
+
+        print(f"Finished writing json file as '{self.write_dir}profile_data.json'.")
+
+def main(stdout_path, write_dir=None):
+    print(f"Path to stdout file: {stdout_path}")
+
+    output_file = open(stdout_path, "r")
+
+    profile_lines = []
+    reached_profile = False
+    print("Parsing stdout file...")
+    count = 0
+    while True:
+        line = output_file.readline()
+
+        if not line:
+            break
+
+        if reached_profile:
+            profile_lines.append(line.strip())
+            count = count + 1
+        elif "TinyProfiler total time across processes" in line:
+            reached_profile = True
+            profile_lines.append(line.strip())
+            count = count + 1
+
+    print(f"Finished parsing stdout file! Found {count} lines from TinyProfiler.")
+
+    full_profile = FullProfile(profile_lines, stdout_path, write_dir)
+
+    full_profile.parse_full_profiling_output()
+    full_profile.write_file()
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2 or len(sys.argv) > 3:
+        raise RuntimeError(f"Usage: {sys.argv[0]} path_to_stdout_file [write_directory]")
+
+    path = sys.argv[1]
+    if len(sys.argv) == 3:
+        write_dir = sys.argv[2]
+        main(path, write_dir)
+    else:
+        main(path)
```

Comment:
This can be improved by using `argparse` (see https://docs.python.org/3/library/argparse.html) in which case you won't need to rely on the order in which arguments are supplied but can instead look for specific arguments (and you can set arguments to be required or optional). This will also allow you to avoid the `if len(sys.argv) == 3:` logic since you can give `write_dir` a default value `None`.

Reply:
Thanks yeah this seems like the place to use that module.